agents-common/src/main/java/org/apache/ranger/plugin/conditionevaluator/RangerTimeOfDayMatcher.java

```diff
@@ -28,176 +34,173 @@
+        LOG.debug("==> RangerTimeOfDayMatcher.init({})", condition);
+
+        super.init();
+
+        if (condition == null) {
+            LOG.debug("init: null policy condition! Will match always!");
+
+            allowAny = true;
+        } else if (CollectionUtils.isEmpty(condition.getValues())) {
+            LOG.debug("init: empty conditions collection on policy condition!  Will match always!");
+
+            allowAny = true;
+        } else {
+            for (String value : condition.getValues()) {
+                if (StringUtils.isEmpty(value)) {
+                    LOG.warn("init: Unexpected: one of the value in condition is null or empty!");
+                } else {
+                    int[] aDuration = extractDuration(value);
+
+                    if (aDuration != null) {
+                        durations.add(aDuration);
+                    }
+                }
+            }
+        }
+
+        if (durations.isEmpty()) {
+            LOG.debug("No valid durations found.  Will always match!");
+
+            allowAny = true;
+        }
+
+        LOG.debug("<== RangerTimeOfDayMatcher.init({}): durations[{}]", condition, this);
+    }
+
+    @Override
+    public boolean isMatched(RangerAccessRequest request) {
+        LOG.debug("==> RangerTimeOfDayMatcher.isMatched({})", request);
+
+        boolean matched = true;
+
+        if (allowAny) {
+            LOG.debug("isMatched: allowAny flag is true.  Matched!");
+        } else if (request == null) {
+            LOG.warn("isMatched: Unexpected: Request is null!  Implicitly matched!");
+        } else if (request.getAccessTime() == null) {
+            LOG.warn("isMatched: Unexpected: Accesstime on the request is null!  Implicitly matched!");
+        } else {
+            Date     date     = request.getAccessTime();
+            Calendar calendar = GregorianCalendar.getInstance();
+
+            calendar.setTime(date);
+
+            int hourOfDay = calendar.get(Calendar.HOUR_OF_DAY);
+            int minutes   = calendar.get(Calendar.MINUTE);
+
+            if (!durationMatched(durations, hourOfDay, minutes)) {
+                matched = false;
+
+                LOG.debug("isMatched: None of the durations contains this hour of day[{}] and minutes[{}]", hourOfDay, minutes);
+            }
+        }
+
+        LOG.debug("<== RangerTimeOfDayMatcher.isMatched({})): {}", request, matched);
+
+        return matched;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+
+        sb.append("RangerTimeOfDayMatcher {");
+        sb.append("allowAny=").append(allowAny).append(" ");
+        sb.append("durations=[");
+        for (int[] duration : durations) {
+            sb.append("{start=").append(duration[0]).append("; end=").append(duration[1]).append("} ");
+        }
+        sb.append("]");
+        sb.append("}");
+
+        return sb.toString();
+    }
+
+    int[] extractDuration(String value) {
```

Comment:
for readability:

```
int[] extractDuration(String value) {
    LOG.debug("==> RangerTimeOfDayMatcher.extractDuration({})", value);

    if (value == null) {
        LOG.warn("extractDuration: null input value!");
        return null;
    }

    Matcher matcher = PATTERN.matcher(value);
    if (!matcher.matches()) {
        LOG.warn("extractDuration: input[{}] did not match pattern!", value);
        return null;
    }

    int startHour = parseHour(matcher.group(1), matcher.group(4));
    int startMinute = parseMinute(matcher.group(3));
    int endHour = parseHour(matcher.group(5), matcher.group(8));
    int endMinute = parseMinute(matcher.group(7));

    int[] result = {(startHour * 60) + startMinute, (endHour * 60) + endMinute};
    LOG.debug("<== RangerTimeOfDayMatcher.extractDuration({}): duration:{}", value, Arrays.toString(result));

    return result;
}

private int parseHour(String hourStr, String period) {
    int hour = Integer.parseInt(hourStr);
    if ("A".equalsIgnoreCase(period) && hour == 12) {
        hour = 0;
    } else if ("P".equalsIgnoreCase(period) && hour != 12) {
        hour += 12;
    }
    return hour;
}

private int parseMinute(String minuteStr) {
    return minuteStr != null ? Integer.parseInt(minuteStr) : 0;
}

```

Reply:
As mentioned earlier, I wouldn't recommend changing from single return to multiple return statements in a method.